File: libraries/leoz-common/src/main/java/org/deku/leoz/messaging/activemq/ActiveMQContext.java
package org.deku.leoz.messaging.activemq;

import org.deku.leoz.messaging.MessagingContext;
import sx.jms.embedded.Broker;
import sx.jms.embedded.activemq.ActiveMQBroker;

import javax.jms.Queue;
import javax.jms.Topic;

/**
 * Messaging context implementation for activemq
 * Created by masc on 16.04.15.
 */
public class ActiveMQContext implements MessagingContext {
    /** Singleton instance */
    private static final ActiveMQContext mInstance = new ActiveMQContext();

    private ActiveMQContext() { }

    public static ActiveMQContext instance() {
        return mInstance;
    }

    @Override
    public Broker getBroker() {
        return ActiveMQBroker.instance();
    }

    @Override
    public Queue getCentralQueue() {
        return this.getBroker().createQueue("leoz.central");
    }

    @Override
    public Queue getCentralEntitySyncQueue() {
        return this.getBroker().createQueue("leoz.entity-sync");
    }

    @Override
    public Queue getCentralLogQueue() {
        return this.getBroker().createQueue("leoz.log");
    }

    @Override
    public Queue getNodeQueue(Integer id) {
        return this.getBroker().createQueue("leoz.node." + id.toString());
    }

    @Override
    public Topic getNodeNotificationTopic() {
        return this.getBroker().createTopic("leoz.notifications");
    }

}

File: libs/sx-mq-jms/src/main/java/sx/mq/jms/activemq/ActiveMQContext.kt
package sx.mq.jms.activemq

import org.apache.activemq.command.ActiveMQQueue
import org.apache.activemq.command.ActiveMQTopic
import sx.mq.jms.JmsContext
import javax.jms.ConnectionFactory
import javax.jms.Queue
import javax.jms.Topic

/**
 * ActiveMQ specific factory
 * Created by masc on 08.05.17.
 */
class ActiveMQContext(
        override val connectionFactory: ConnectionFactory
) : JmsContext {
    override fun createQueue(name: String): Queue {
        return ActiveMQQueue(name)
    }

    override fun createTopic(name: String): Topic {
        return ActiveMQTopic(name)
    }
}